### PR TITLE
:seedling: Add release-1.8 configurations

### PR DESCRIPTION
**Pull request by mquhuy**
Updates the prow config and README for CAPM3 and IPAM release-1.8 configurations.

### CODE REVIEW

**Review by smoshiur1237 — COMMENTED** [review]
LGTM

File: prow/manifests/overlays/metal3/config.yaml [hunk 1]
```diff
@@ -960,17 +966,12 @@ presubmits:
...
-    - release-0.4
-    agent: jenkins
-    always_run: false
-    optional: true
 
   metal3-io/cluster-api-provider-metal3:
   - name: gomod
     branches:
     - main
+    - release-1.8
```
Comment (NymanRobin) [inline prow/manifests/overlays/metal3/config.yaml:974]: The definition for branch release-1.5 could be removed now for the affected jobs, right? That would reduce this massive config a little bit :smile:

File: prow/manifests/overlays/metal3/config.yaml [hunk 2]
```diff
@@ -2587,59 +2616,59 @@ presubmits:
...
     branches:
-    - release-1.5
+    - release-1.7
     agent: jenkins
     always_run: false
     optional: true
-  # name: {job_prefix}-e2e-clusterctl-upgrade-test-{capm3_target_branch}
-  - name: metal3-e2e-clusterctl-upgrade-test-main
+  # name: {job_prefix}-e2e-{k8s_versions}-upgrade-test-{capm3_target_branch}
+  - name: metal3-e2e-1-28-1-29-upgrade-test-main
```
Comment (NymanRobin) [inline prow/manifests/overlays/metal3/config.yaml:2639]: Please double check what versions are going to be tested in the new main upgrade

**Review by NymanRobin — COMMENTED** [review]
I think this needs to rebased as a consequence of this pr #838 

Also for the upgrade tests the kubernetes version needs to be lifted most likely

File: prow/manifests/overlays/metal3/config.yaml (hunk 1 above)
Comment (mquhuy) [reply prow/manifests/overlays/metal3/config.yaml:974]: You're probably right, but I left them there since I wasn't sure :sweat_smile:

Comment (tuminoid) [reply prow/manifests/overlays/metal3/config.yaml:908]: Do adding 1.8 and removing 1.5 in two different PRs please. Much easier to review.

**Review by tuminoid — COMMENTED** [review]
/approve

**Comment by tuminoid** [comment]
> Since this is just not adding 1.8 but also regrouping, make note of that in title and description as well. Its good that they're in different commits, the best would've been different PRs.

Actually there is another commit doing the regrouping and this is just reusing it. Hence this needs rebase after its merged.
/hold

**Comment by mquhuy** [comment]
/unhold
The other PR was merged and this one was rebased.

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Pull request merged**